src/pin.rs

```diff
@@ -187,15 +244,73 @@ impl InputPin for Mock {
     }
 }
 
+impl PwmPin for Mock {
+    type Duty = PwmDuty;
+
+    fn disable(&mut self) {
+        let Transaction { kind, err: _err } =
+            self.next().expect("no expectation for pin::disable call");
+
+        assert_eq!(kind, TransactionKind::Disable, "expected pin::disable");
+    }
+
+    fn enable(&mut self) {
+        let Transaction { kind, err: _err } =
+            self.next().expect("no expectation for pin::enable call");
+
+        assert_eq!(kind, TransactionKind::Enable, "expected pin::enable");
+    }
+
+    fn get_duty(&self) -> Self::Duty {
+        let mut s = self.clone();
+
+        let Transaction { kind, err: _err } =
+            s.next().expect("no expectation for pin::get_duty call");
+
+        if let TransactionKind::GetDuty(duty) = kind {
+            duty
+        } else {
+            assert!(false, "expected pin::get_duty");
+            unreachable!()
+        }
+    }
+
+    fn get_max_duty(&self) -> Self::Duty {
+        let mut s = self.clone();
+
+        let Transaction { kind, err: _err } =
+            s.next().expect("no expectation for pin::get_max_duty call");
+
+        if let TransactionKind::GetMaxDuty(max_duty) = kind {
+            max_duty
+        } else {
+            assert!(false, "expected pin::get_max_duty");
+            unreachable!()
```

Comment:
Instead of `assert!(false, msg)` followed by `unreachable!()`, simply use `panic!(msg)`. It's the same.

```suggestion
            panic!("expected pin::get_max_duty");
```

Same thing counts for other methods as well.

Reply:
Yeah, Rust's assertions are not too fancy unfortunately 🙂